src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -347,6 +397,13 @@ public synchronized void commit() throws IOException {
         while (streamsToFlush.size() > 1) {
             streamsToFlush.removeFirst().close();
         }
+
+        // Roll the log file if we exceed the size limit
+        long logSize = getCurrentLogSize();
```

Comment:
Please change the order of the if conditions. You probably don't want to calculate the current log file size if the feature is disabled. Probably not a big deal performance-wise, but I think it's cleaner code.